### PR TITLE
Session5 HW

### PR DESCRIPTION
**Pull request by celestfradel**
session 5 HW. For Review. Thank you

### CODE REVIEW

File: Repositories/ItemRepository.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        public void Delete(int id)
+        {
+            var itemToRemove = _dataContext.Items.SingleOrDefault(x => x.Id == id);
+            if (itemToRemove != null)
+                _dataContext.Items.Remove(itemToRemove);
+        }
+
+        public void Save(Item item)
+        {
+            _dataContext.Items.Add(item);
```
Comment (jegtugado) [inline Repositories/ItemRepository.cs:31]: Can you modify this to be able to handle both add and update scenarios?

File: Services/ItemService.cs [hunk 2]
```diff
@@ -1,35 +1,83 @@
...
+            var item = _mapper.Map<Item>(itemDto);
+            _itemRepository.Save(item);
         }
 
-        public void Save(ItemDto model)
+        public void Update(ItemDto itemDto)
         {
-            //Maps ItemDTO to Item Class
-            var item = _mapper.Map<Item>(model);         
+            var itemToUpdate = _itemRepository.All().Where(x => x.Id == itemDto.Id).SingleOrDefault();
```
Comment (jegtugado) [inline Services/ItemService.cs:79]: please use IItemRepository.Save method for updating

File: ASPNetCoreMasters.Api/Startup.cs [hunk 3]
```diff
@@ -28,6 +29,8 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddAutoMapper(typeof(MappingProfile));
             services.AddControllers();
+            services.AddScoped<DataContext>();
```
Comment (jegtugado) [inline ASPNetCoreMasters.Api/Startup.cs:32]: We recommend using Singleton lifetime for the datacontext so you can verify if the Item Add or Update worked

**Review by jegtugado — CHANGES_REQUESTED** [review]
Hi sir, minor changes needed.

File: ASPNetCoreMasters.Api/Startup.cs (hunk 3 above)
Comment (celestfradel) [reply ASPNetCoreMasters.Api/Startup.cs]: updated Session5 HW for the changes. For Review

**Review by jegtugado — APPROVED** [review]

**Pull request merged**